cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/internal/BootstrapCoordinator.scala

```diff
@@ -182,7 +182,11 @@ private[akka] class BootstrapCoordinator(discovery: ServiceDiscovery,
             case (host, immutable.Seq(singleResult)) =>
               immutable.Seq(singleResult)
             case (host, multipleResults) =>
-              multipleResults.filter(_.port.contains(settings.contactPoint.fallbackPort))
+              if (multipleResults.exists(_.port.isDefined)) {
+                multipleResults.filter(_.port.contains(settings.contactPoint.fallbackPort))
+              } else {
+                multipleResults
```

Comment:
> I personally would have changed it to change every None port to the fallback-port. As that would be the expected behavior IMHO.

I think you're right. I can't think of any reason to return multiple results for the same host with no ports. The fallbackport will be added to each of them and we'll probe the same node multiple times.

First there's the easy case:

* The user has defined a port in their query so it is up to service discovery impl to return only those with that port

Then for when a user does not include a port in the query:

* Discovery returns multiple targets for the same host, without ports. We should fill in the fallback port
* Discovery returns multiple targets with different ports: right now we filter based on the fallback port as we're assuming that they are mutiple ports for the same service e.g. remoting port, management port, HTTP port. This makes bootstrap easier to use with less configuration (not having to specify the port name). But it breaks the case where dynamic ports are being used for management so all the targets should be kept. 

For the last point I can't think of a way to support both without adding another configuration flag saying that management has dynamic ports so don't filter.

WDYT @milanvdm

Reply:
I fully agree with that approach.

Although Im not sure if your last point needs an extra configuration flag.

We get the following list:
```
host1:None
host1:1234
host2:None
host2:8558
host2:1234
```

You add the `fallback-port` to the `None`. Would you expect the `host1` not to show up in the final list of hosts?
For me, here again, whenever you have a `None`, the `fallback-port` is used. This would cause the `host1` to get into the final list of hosts.

Reply:
For your example I think there two behaviors we want to support.

* Replace the Nones with fallback ports and return only the results that either contained None or are the same as the fallback port. This is the scenario where all akka management ports are the same across your cluster. This is nearly always the case in Kubernetess for example.
* Replace the Nones with fallback ports and don't do any filtering apart from maybe duplicates.  This is the case where the akka management port is dynamic and we can't assume that results with the same host and a different port are the same Akka nodes as they could be different Akka nodes on the same host with different akka management ports.

It is the latter I think is broken atm.

Reply:
Understood. 
If that is indeed something that needs to be supported, then you need the extra configuration.

Do you add this to this PR or is this something I need to do as an extra?

Reply:
I can follow up with another PR

Reply:
Note that we already fill in ports that are None with the fallback port: https://github.com/akka/akka-management/blob/master/cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/internal/BootstrapCoordinator.scala#L286